grails-app/services/com/unifina/service/PermissionService.groovy

```diff
@@ -139,6 +138,14 @@ class PermissionService {
 			}
 		}.toList()
 
+		if (resource instanceof Stream && resource.uiChannel && resource.uiChannelCanvas != null && resource.uiChannelPath != null) {
```

Comment:
Is this basically the same condition as line `150`? They could be combined. 

(This one has some additional conditions which we can keep, but basically `uiChannelCanvas` and `uiChannelPath` should always be set if `resource.uiChannel` is true.)